src/main/flight/imu.c

```diff
@@ -110,6 +132,11 @@ STATIC_UNIT_TESTED void imuComputeRotationMatrix(void)
     rMat[2][0] = 2.0f * (q1q3 + -q0q2);
     rMat[2][1] = 2.0f * (q2q3 - -q0q1);
     rMat[2][2] = 1.0f - 2.0f * q1q1 - 2.0f * q2q2;
+
+#if defined(SIMULATOR_BUILD) && defined(SKIP_IMU_CALC) && !defined(SET_IMU_FROM_EULER)
+    rMat[1][0] = -2.0f * (q1q2 - -q0q3);
```

Comment:
This seems to be caused by different orientation systems used in real and simulator build. Hacking it this way is very bad idea, at least in long-term.
It also seems that you are fixing only part of matrix which is used in code, rest of entries are probably wrong

I'll try to get gazebo8 (not that simple on debian) and look into this